### ISSUE TITLE
Flash Fill auto-complete

### ISSUE THREAD

**Issue by jzabroski** [dotnet/command-line-api#439]
Is this team familiar with the Microsoft Research Prose SDK?

**Comment by jonsequitur** [comment]
I just took a quick peek at it. Very interesting stuff. Can you expand on how it relates to auto-complete?

**Comment by jzabroski** [comment]
Yeah. Let's say you always do a series of commands, one after the other. It would be ideal if the command line was intelligent and recognized your coding patterns, the same way Excel recognizes Flash Fill opportunities.

One way to imagine this (I'm not a UX specialist) would be to think of how auto-complete works in cshell, where when you TAB twice on something that has many matches, it outputs the first N matches lexicographically.  bash also works this way, but Bill Joy to my knowledge invented the concept.

In the context of our discussion on benchmarkDotNet, there are probably some auto-complete flash fill options that make sense, as if you were choosing from a dropdown.  And the ordering of those options might be made more optimal based on your recently chosen arguments to those parameters, rather than say lexicographical ordering.

**Comment by jonsequitur** [comment]
The System.CommandLine suggestions model is open-ended and a feature like this could most likely be built with it. Middleware could capture the inputs to train the model and a custom `ISuggestionSource` could read from the model.
